src/garmin.c

```diff
@@ -322,7 +322,8 @@ mtp_get_file_list(dc_device_t *abstract, struct file_list *files)
 	free(rawdevices);
 	DEBUG(abstract->context, "Found %d files", files->nr);
 
-	qsort(files->array, files->nr, sizeof(struct fit_file), name_cmp);
+	if (files->array)
+		qsort(files->array, files->nr, sizeof(struct fit_file), name_cmp);
```

Comment:
Does this matter?

If the number of entries is zero, there should never be any access to the array pointer. So the test for NULL seems entirely unnecessary.

Reply:
It's not entirely clear from the standard if `qsort()` might dereference the pointer if the second argument is zero.
So this seems like very cheap insurance.

Reply:
but I guess then I should check in `get_file_list()` as well?
meh